apps/percona/functional/snapshot_rebuild/test.yml

```diff
@@ -0,0 +1,189 @@
+---
+## Test involved.
+#Check if the application is running already.
+#Selct one of the replicas.
+#Introduce packet drop in the selected replica.
+#Create snapshot.
+#check if the snapshot is not present in the degraded pod.
+#Bring up the replica.
+#Check if the snapshot is rebuilt.
+
+- hosts: localhost
+  connection: local
+
+  vars_files:
+    - test_vars.yml
+
+  tasks:
+    - block:
+
+      ## Generating the testname.
+        - include_tasks: /common/utils/create_testname.yml
+
+         ## RECORD START-OF-TEST IN LITMUS RESULT CR
+        - include_tasks: "/common/utils/update_litmus_result_resource.yml"
+          vars:
+            status: 'SOT'
+
+         ## Check if the application is running.
+        - name: Checking if the application is up and running.
+          shell: kubectl get pods -n {{ namespace }} -l {{ application_label }} --no-headers -o custom-columns=:status.phase
+          args:
+            executable: /bin/bash
+          register: app_status
+          until: "'Running' in app_status.stdout"
+          delay: 10
+          retries: 3
+
+        - name: Derive PV name from PVC
+          shell: >
+            kubectl get pvc {{ app_pvc }} -n {{ namespace }}
+            --no-headers -o custom-columns=:spec.volumeName
+          args:
+            executable: /bin/bash
+          register: pv
+
+        - name: Derive the cstor target using pv.
+          shell: >
+            kubectl get pods -n {{ operator_ns }} -l openebs.io/persistent-volume={{ pv.stdout }} 
+            -l openebs.io/target=cstor-target -o custom-columns=:metadata.name --no-headers
+          args:
+            executable: /bin/bash
+          register: cstor_target
+
+        - name: Obtaining the consistency factor of cstor target.
+          shell: >
+            kubectl exec -it {{ cstor_target.stdout }} -n {{ operator_ns }} --container cstor-istgt 
+            -- cat /usr/local/etc/istgt/istgt.conf | grep -A8 "{{ pv.stdout }}" | grep ConsistencyFactor| cut -d ' ' -f 4
+          args:
+            executable: /bin/bash
+          register: consistency_factor
+
+        - name: Checking the number of healthy pods.
+          shell: >
+            kubectl exec -it {{ cstor_target.stdout }} -n {{ operator_ns }} --container  cstor-istgt 
+            -- istgtcontrol -q replica |json_pp |grep "\"replicaId" |awk -F ':' '{print $2}' | tr -d ',' |wc -l
+          args:
+            executable: /bin/bash
+          register: healthy_pods
```

Comment:
below is the output of istgtcontrol -q replica:
```{
  "volumeStatus": [
    {
      "replicaStatus": [
        {
          "upTime": 30885,
          "inflightWrite": "0",
          "replicaId": "9630529016344247398",
          "status": "Healthy",
          "checkpointedIOSeq": "0",
          "inflightSync": "0",
          "inflightRead": "0"
        },
        {
          "checkpointedIOSeq": "0",
          "inflightSync": "0",
          "inflightRead": "0",
          "replicaId": "12636181477300307024",
          "inflightWrite": "0",
          "status": "Healthy",
          "upTime": 30884
        },
        {
          "status": "Healthy",
          "inflightWrite": "0",
          "replicaId": "17061940481940744402",
          "upTime": 30882,
          "inflightRead": "0",
          "inflightSync": "0",
          "checkpointedIOSeq": "0"
        }
      ],
      "status": "Healthy",
      "name": "pvc-2d039cc0-f782-11e8-ba31-42010af00275"
    }
  ]
}
```
To get healthy_pods, better to do something like
` -- istgtcontrol -q replica |json_pp | jq '.volumeStatus[0].replicaStatus[].status' | grep Healthy | wc -l`

Reply:
also, we need to make sure that healthy_pods is greater than consistency factor to take snapshots

Reply:
or this can be checked to be equal to replication factor instead of consistency factor so that next tasks will go fine

Reply:
@vitta, We have this check already implemented . Snapshot creation will be performed only if the healthy replicas are greater than or equal to consistency factor.